client/scripts/views/pages/new_proposal/new_proposal_form.ts

```diff
@@ -341,28 +353,40 @@ const NewProposalForm = {
         vnode.state.proposer = app.user?.activeAccount?.address;
         if (!vnode.state.proposer) throw new Error('Invalid address / not logged in');
         if (!vnode.state.executor) throw new Error('Invalid executor');
-        if (!vnode.state.targets) throw new Error('No targets');
-        if (!vnode.state.values) throw new Error('No values');
-        // signatures optional
-        if (!vnode.state.signatures) {
-          vnode.state.signatures = '';
-        }
-        if (!vnode.state.calldatas) throw new Error('No calldatas');
-        if (!vnode.state.withDelegateCalls) throw new Error('No withDelegateCalls');
         if (!vnode.state.ipfsHash) throw new Error('No ipfs hash');
 
-        // TODO: support multiple calls
-        const targets: string[] = [ vnode.state.targets ];
-        const values: string[] = [ vnode.state.values ];
-        const calldatas: string[] = [ vnode.state.calldatas ];
-        const signatures: string[] = [ vnode.state.signatures ];
-        const withDelegateCalls: boolean[] = [ vnode.state.withDelegateCalls ]
-          .map((v: string): boolean => {
-            if (v.trim().toLowerCase() === 'true') return true;
-            if (v.trim().toLowerCase() === 'false') return false;
-            throw new Error(`invalid withDelegateCalls string: ${v}`);
-          });
+        const targets = [];
+        const values = [];
+        const calldatas = [];
+        const signatures = [];
+        const withDelegateCalls = [];
 
+        for (let i = 0; i < vnode.state.aaveTabCount; i++) {
```

Comment:
The contract requires that all these arrays are the same length, even if the optional argument is omitted. So we should be pushing empty strings if unset, rather than just ignoring it. This is because for the first target, it will look at the first element of values/calldatas/signatures/withdelegatecalls, the second target to the second element of each, and so forth.